src/quest/manager.py

```diff
@@ -191,6 +218,30 @@ async def _deregister_alias(self, alias: str):
         if alias in self._alias_dictionary:
             del self._alias_dictionary[alias]
 
+    def get_workflow_metrics(self):
+        """Return metrics for active workflows"""
+        metrics = []
+        for wtype, wid, args, kwargs, background, start_time in self._workflow_data:
+            metrics.append({
+                "workflow_id": wid,
+                "workflow_type": wtype,
+                "start_time": start_time
+            })
+        return metrics
+
+    # Should not be used for background tasks
+    async def get_workflow_result(self, workflow_id: str, delete: bool = True):
+        if workflow_id not in self._futures:
+            return None
+
+        # Wait for the future to complete and return result
+        future = self._futures[workflow_id]
+        result = await future
+        if delete and workflow_id in self._results:
+            del self._results[workflow_id]
+            self._futures.pop(workflow_id)
+        return result
```

Comment:
We need to make sure the results and futures get deleted, even if a future raises an exception:

```py
try:
    return await future
finally:
    if delete:
        del self._futures[workflow_id]  # Using del in this case should also fix the await warning
        if workflow_id in self._results:
            del self._results[workflow_id]
```